ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -485,6 +480,26 @@ private void clearLocksAndHB() {
     stopHeartbeat();
   }
 
+  private void cleanupDirForCTAS() {
+    if (HiveConf.getBoolVar(conf, HiveConf.ConfVars.TXN_CTAS_X_LOCK)) {
+      if (destinationTable != null) {
+        try {
+          CompactionRequest rqst = new CompactionRequest(
+                  destinationTable.getDbName(), destinationTable.getTableName(), CompactionType.MAJOR);
+          rqst.setRunas(TxnUtils.findUserToRunAs(destinationTable.getSd().getLocation(),
+                  destinationTable.getTTable(), conf));
+
+          rqst.putToProperties("location", destinationTable.getSd().getLocation());
+          rqst.putToProperties("ifPurge", Boolean.toString(true));
+          TxnStore txnHandler = TxnUtils.getTxnStore(conf);
+          txnHandler.submitForCleanup(rqst, destinationTable.getTTable().getWriteId(), getCurrentTxnId());
+        } catch (InterruptedException | IOException | MetaException e) {
+          throw new RuntimeException("Not able to submit cleanup operation of directory written by CTAS");
```

Comment:
should we catch just InterruptedException | IOException and re-throw MetaException here?

Reply:
All three exceptions are thrown within the code. Changed it to LockException instead of RuntimeException.